scipy/stats/_distribution_infrastructure.py

```diff
@@ -3527,16 +3621,29 @@ def make_distribution(dist):
     _x_support = _RealDomain(endpoints=support, inclusive=(True, True))
     _x_param = _RealParameter('x', domain=_x_support, typical=(-1, 1))
 
+    repr_str = _distribution_names.get(dist.name, dist.name.capitalize())
+
     class CustomDistribution(ContinuousDistribution):
         _parameterizations = ([_Parameterization(*parameters)] if parameters
                               else [])
         _variable = _x_param
 
+        def __repr__(self):
+            s = super().__repr__()
+            return s.replace('CustomDistribution', repr_str)
+
+    # override the domain's `get_numerical_endpoints` rather than the
+    # distribution's `_support` to ensure that `_support` takes care
+    # of any required broadcasting, etc.
+    def get_numerical_endpoints(parameter_values):
+        a, b = dist._get_support(**parameter_values)
+        return np.asarray(a)[()], np.asarray(b)[()]
+
     def _sample_formula(self, _, full_shape=(), *, rng=None, **kwargs):
         return dist._rvs(size=full_shape, random_state=rng, **kwargs)
 
-    def _moment_raw_formula(self, n, **kwargs):
-        return dist._munp(int(n), **kwargs)
+    def _moment_raw_formula(self, order, **kwargs):
+        return dist._munp(int(order), **kwargs)
```

Comment:
This should have been `order` like everwhere else. This also gave `irwinhall` some trouble because `n` is a shape parameter.

Currently, `irwinhall` doesn't work with `order_statistic` because of the name conflict. We might want to rename the shapes of `OrderStatisticDistribution`, or we can just address this when we address the name conflict issue in general.